jax/lax/lax.py

```diff
@@ -4022,6 +4035,23 @@ def _tie_in_batch_rule(batched_args, batch_dims):
 masking.masking_rules[tie_in_p] = lambda vals, logical_shapes: vals[1]
 
 
+push_scope_p = Primitive('push_scope')
+push_scope_p.def_impl(lambda scope, name: tagging.Scope(scope.path + (name,)))
+push_scope_p.def_abstract_eval(lambda x, name: tagging.AbstractScope(x.path + (name,)))
+xla.translations[push_scope_p] = lambda c, x, name: x
+
+
+tag_p = Primitive('tag')
+tag_p.def_impl(lambda x, scope: x)
+tag_p.def_abstract_eval(lambda x, scope: x)
+xla.translations[tag_p] = lambda c, x, scope: x
```

Comment:
How does tagging work under JIT? I see from the tests that it does work, but how?